### ISSUE TITLE
Thumbnails from Session not rendered

### ISSUE THREAD

**Issue by luki21213** [FineUploader/react-fine-uploader#91]
When loading uploads (images) from session, every image appears as shown in the image below.
![image](https://cloud.githubusercontent.com/assets/10244946/24460532/db71013e-149e-11e7-9d95-5a65d0c5a4df.png)

When adding a timeout around the drawThumbnail() function call in react-fine-uploader everything seems to be fine.
`setTimeout(() => this.props.uploader.methods.drawThumbnail(...), 200)`

It seems that drawThumbnail tries to access the thumbnailUrl even before it is there.

Is this a bug in react-fine-uploader or am I missing something?

Thanks :-)

#update (my code)

```
import React, { Component } from 'react'
import cookie from 'react-cookie'
import Gallery from 'react-fine-uploader'
import FineUploaderTraditional from 'fine-uploader-wrappers'
import 'react-fine-uploader/gallery/gallery.css'

const authToken = cookie.load('angelAppToken')
const uploader = new FineUploaderTraditional({
  options: {
    session: {
      endpoint: 'http://localhost:9091/api/rest/uploads',
      customHeaders: {
        'authorization': `Bearer ${authToken}`
      },
      params: {
        angelOrderId: 4083,
        tags: ['unwashed']
      }
    },
    debug: true,
    request: {
      endpoint: 'http://localhost:9091/api/rest/uploads',
      params: {
        angelOrderId: 4083,
        tag: ['unwashed']
      },
      customHeaders: {
        'authorization': `Bearer ${authToken}`
      }
    },
    retry: {
      enableAuto: true
    }
  }
})

export class Images extends Component {
  render() {
    return (
      <Gallery uploader={ uploader } />
    )
  }
}

export default Images
```

http://localhost:9091/api/rest/uploads returns an array like

```
[
  {
    "name":"058.jpg",
    "size":834793,
    "thumbnailUrl":"someUrl",
    "uuid":259,
    "deleteFileEndpoint":"someUrl",
    "deleteFileParams":""
  }
]
```

the setTimeout i made for debugging was in the react-fine-uploader node module

**Comment by rnicholus** [comment]
Please show _all_ of your code so this can be easily reproduced/investigated.

**Comment by rnicholus** [comment]
>When adding a timeout around the drawThumbnail() function call 

Did you make this change locally, in the `<Thumbnail />` component?

**Comment by rnicholus** [comment]
confirmed - this will have to be fixed before 1.0.0. if you're able to fix and open up a PR, please do let me know. otherwise, i'll get to this when i have a chance.

**Comment by rnicholus** [comment]
Upon closer inspection, this is somewhat of a "bug" in Fine Uploader itself. The thumbnail URL is not stored with the file until a few lines after the file is added to the internal store and the `UPLOAD_SUCCESSFUL` status change event is fired. So, wrapping `drawThumbnail` in a `setTimeout` of even 1 ms is sufficient to workaround this particular issue. However, this seems like a kludge, and it probably makes sense to "fix" this in Fine Uploader by ensuring the totality of the canned file record is processed _before_ the `UPLOAD_SUCCESSFUL` status change event is broadcast.

**Comment by rnicholus** [comment]
Fixed in React Fine Uploader v1.0.0-rc2. Can you confirm that this fixes the issue for you? Please read the release notes for this version first at https://github.com/FineUploader/react-fine-uploader/releases/tag/1.0.0-rc2. There is a breaking change.

**Comment by luki21213** [comment]
Thank you this release fixed the issue!
